README.md

```diff
@@ -0,0 +1,284 @@
+# nurbs
+
+> [Non-Uniform Rational B-Splines][nurbs] (NURBS) of any dimensionality
+
+This library implements n-dimensional [Non-Uniform Rational B-Splines][nurbs] (NURBS). It has no dependencies and uses code generation to unroll loops, optimize for various cases (uniform and non-uniform; rational and non-rational; clamped, open, and periodic) and allow compatibility with multiple input types (arrays of arrays, [ndarrays][ndarray]).
+
+## Installation
+
+**Not yet published.**
+
+```bash
+$ npm install nurbs
+```
+
+## Examples
+
+### Open B-Spline
+
+To construct an open quadratic B-Spline in two dimensions:
+<!--The following example, instead of creating a new spline, this example updates the existing spline by calling the existing spline as a constructor, which re-runs the validations while preserving the same instance. (This feature may be removed. It's simple but weird and since the splines have very little internal state, it really doens't save much speed or allocation.) In general, references to the input data are retained so that you may mutate the data (including the number of points) after instantiation. If you change the degree or dimensionality of the curve though, you must either create a new curve or invoke the existing curve as a constructor and pass new data.-->
```

Comment:
This comment isn't visible but I think that's intentional

Reply:
Yeah, bothered to write it but didn't have a place for it. Or really, the readme just got too long. Suggests maybe API.md and examples.md would be appropriate.